src/ui/VirtualFileRenderer/ColorBar.tsx

```diff
@@ -0,0 +1,44 @@
+interface CoverageBarProps {
+  locationHash?: string
+  lineNumber: number
+  coverage?: 'H' | 'M' | 'P'
+}
+
+// exporting for testing purposes
+export const ColorBar = ({
+  coverage,
+  locationHash,
+  lineNumber,
+}: CoverageBarProps) => {
+  if (locationHash && locationHash === `#L${lineNumber}`) {
+    return (
+      <div
+        data-testid="highlighted-bar"
+        className="pointer-events-none absolute left-[-72px] h-full w-[calc(100%+72px)] bg-ds-blue-medium opacity-25"
+      />
+    )
+  } else if (coverage === 'M') {
+    return (
+      <div
+        data-testid="uncovered-bar"
+        className="pointer-events-none absolute left-[-72px] h-full w-[calc(100%+72px)] bg-ds-coverage-uncovered opacity-25"
+      />
+    )
+  } else if (coverage === 'P') {
+    return (
+      <div
+        data-testid="partial-bar"
+        className="pointer-events-none absolute left-[-72px] h-full w-[calc(100%+72px)] bg-ds-coverage-partial opacity-25"
+      />
+    )
+  } else if (coverage === 'H') {
+    return (
+      <div
+        data-testid="covered-bar"
+        className="pointer-events-none absolute left-[-72px] h-full w-[calc(100%+72px)] bg-ds-coverage-covered opacity-25"
+      />
+    )
+  } else {
+    return null
```

Comment:
This is hiding a future bug. You should never do this when dealing with an exhaustive type and should always throw an error.

Reply:
So the type can be either highlighted, one of the three coverage values, or undefined, and when it's undefined we don't want this to be rendered, so this case is what we want.

Reply:
You should add a line at the start to cover undefined then and a captureMessage to detect any other values. Right now if someone adds a new coverage type, you will need to keep track of all the areas where this is defined and update it, which is easy to miss and forget to update. Instead you should do something like

```
function exhaustive(_x: never): void {}

function coverage(type: "A"|"B"|undefined)
if(type === undefined) return // valid

switch(type){
  case "A":
    return value
  case "B":
    return other value
   default: {
     Sentry.captureMessage(`Invalid coverage value ${type}`);
     exhaustive(type)
   }
}
```

Reply:
What is this `exhaustive` function for?

Reply:
Ahh I think it's just a placeholder for this example ... i think ... I can just return `null` as the default return and capture the message beforehand.

Reply:
Exhaustive check is to ensure that you are handling each case. If you for example remove the B case, then the type you are trying to pass to exhaustive will be "B", which will fail. If you handle B, then the type is narrowed to `never`, because it cannot happen

Reply:
Ahhh okay noice, done